File: 프로그래머스/lv2.소수찾기.py
# Level2: 소수찾기
from itertools import permutations
import math
def solution(numbers):
    answer = 0
    def isPrime(num):
        if num < 2:
            return False
        for i in range(2,int(math.sqrt(num)+1)):
            if num % i == 0:
                return False
        return True
    candidates = set()
    for i in range(1,len(numbers)+1):
        tmp = list(permutations(numbers,i))
        for t in tmp:
            candidates.add(int("".join(t)))
            
    for c in candidates:
        if isPrime(c):
            answer += 1
    return answer

File: 프로그래머스/lv2.소수찾기.js
const getPermutations = function(arr, selectNumber){
    const result = [];
    if(selectNumber === 1) return arr.map(ele => [ele]);
    arr.forEach((fixed,idx,origin) => {
		const rest = [...origin.slice(0,idx),...origin.slice(idx+1)];
		const permutaions = getPermutations(rest,selectNumber-1);
		const attached = permutaions.map((element) => [fixed,...element]);
		result.push(...attached);
		}
    );
    return result;
}
const isPrime = function(number){
    if(number <= 1) return false;
    for(let i = 2; i <= parseInt(Math.sqrt(number)); i++){
        if(number % i === 0) return false;
    }
    return true;
}
function solution(numbers) {
    let answer = 0;
    let candidates = new Set();
    const number = numbers.split("");
    for(let i = 1; i <= numbers.length; i++){
        const tmp = getPermutations(number,i);
        tmp.forEach((ele) => candidates.add(parseInt(ele.join(""))));
    }
    candidates.forEach((ele) => {
        if(isPrime(ele)) answer += 1;
           });
    return answer;
}